src/NumericalAlgorithms/LinearOperators/Divergence.hpp

```diff
@@ -0,0 +1,51 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions computing partial derivatives.
+
+#pragma once
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataBoxTag.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+
+/// \ingroup NumericalAlgorithmsGroup
+/// \brief Compute the (Euclidean) divergence of fluxes
+///
+/// \return a `Variables` with the same structure as `F` and each `Tag` wrapped
+/// with a `Tags::div`.
+template <typename FluxTags, size_t Dim, typename DerivativeFrame>
+auto divergence(
+    const Variables<FluxTags>& F, const Index<Dim>& extents,
+    const Tensor<DataVector, tmpl::integral_list<std::int32_t, 2, 1>,
+                 typelist<SpatialIndex<Dim, UpLo::Up, Frame::Logical>,
+                          SpatialIndex<Dim, UpLo::Lo, DerivativeFrame>>>&
+        inverse_jacobian) noexcept {
```

Comment:
Return type can't be auto for compute item. I used trailing return type in partial_derivatives

Reply:
Does it need to be trailing? The return type is `Variables<db::wrap_tags_in<Tags::div, FluxTags>>` which I thought would be deducible.

Reply:
I don't think it has to be, no, either way is fine to me

Reply:
please clarify: auto is okay, or I need to explicitly give the return type?

Reply:
You need to specify the return type

Reply:
This is now visible in #415 but there is a discussion about where to put the tags and whether or not to generate explicit instantiations functions that take Variables